### ISSUE TITLE
Understand the sub-commodity and fit the choice model

### ISSUE THREAD

**Issue by YuXia-SR** [YuXia-SR/dunnhumby_analysis#3]
Let's move to the interesting part - model fitting. Please refer to this [paper](https://cseweb.ucsd.edu/~jmcauley/pdfs/www17.pdf) for model structure. The desired output for this issue is the fitted model to tell us if a customer will make a purchase on a specific sub-commodity group at a given time. 

- [ ] Output a description table about the size, revenue of all sub-commodity groups.
- [ ] Define function to extract the sub-commodity level features from the filtered transaction data. For the feature list, please refer to the table 3 in the paper above.
- [ ] Define a model class that inherits from the scikit-learn [BaseEstimator](https://scikit-learn.org/stable/modules/generated/sklearn.base.BaseEstimator.html), [ClassifierMixin](https://scikit-learn.org/stable/modules/generated/sklearn.base.ClassifierMixin.html).
- [ ] Fit the model with the transaction data.
- [ ] Evaluate the model and output an accuracy report.
